src/Error/Middleware/ErrorHandlerMiddleware.php

```diff
@@ -244,4 +259,30 @@ protected function getExceptionTrap(): ExceptionTrap
 
         return $this->exceptionTrap;
     }
+
+    /**
+     * Ensure that the application's routes are loaded.
+     *
+     * @return void
+     */
+    protected function loadRoutes(): void
+    {
+        if (
+            !($this->app instanceof RoutingApplicationInterface)
+            || Router::routes()
+        ) {
+            return;
+        }
+
+        try {
+            $builder = Router::createRouteBuilder('/');
+
+            $this->app->routes($builder);
+            if ($this->app instanceof PluginApplicationInterface) {
+                $this->app->pluginRoutes($builder);
+            }
+        } catch (Throwable $e) {
+            // Continue to avoid recursive exceptions.
```

Comment:
Should we trigger a warning or log here?

Reply:
I guess we could trigger a warning, wouldn't want to fill up the logs with nested errors.